### PR TITLE
Add FXIOS-5821 [v112] Notification opt-in onboarding card

### PR DESCRIPTION
**Pull request by thatswinnie**
### [FXIOS-5821](https://mozilla-hub.atlassian.net/browse/FXIOS-5821) | #13306 

This adds the notification onboarding card including the experiment for it's position.

### CODE REVIEW

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Project coverage: 30.18%</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Edited 18 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 3 files</td>
    </tr>
  </tbody>
</table>

## Client.app: Coverage: 27.1
| File | Coverage ||
| --- | --- | --- |
SyncContentSettingsViewController.swift | 0.0% | ⚠️
FxAWebViewController.swift | 0.0% | ⚠️
FirefoxAccountSignInViewController.swift | 0.0% | ⚠️
FeatureFlagsManager.swift | 100.0% | ✅
NimbusFlaggableFeature.swift | 99.19% | ✅
FxAWebViewModel.swift | 0.0% | ⚠️
OnboardingNotificationCardHelper.swift | 25.0% | ⚠️
OnboardingCardViewController.swift | 0.0% | ⚠️
NimbusFeatureFlagLayer.swift | 87.08% | ✅
IntroViewController.swift | 0.0% | ⚠️
IntroViewModel.swift | 67.86% | ✅
UpdateViewModel.swift | 71.58% | ✅

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against db011ae49706d12e84fa78c9d02538ed7b78f287
</p>

File: Client/Frontend/Intro/IntroViewModel.swift [hunk 1]
```diff
@@ -46,7 +49,16 @@ struct IntroViewModel: OnboardingViewModelProtocol, FeatureFlaggable {
     }
 
     var enabledCards: [IntroViewModel.InformationCards] {
-        return [.welcome, .signSync]
+        let notificationCardPosition = OnboardingNotificationCardHelper().cardPosition
```
Comment (yoanarios) [inline Client/Frontend/Intro/IntroViewModel.swift:52]: Can we add unit test for this logic?

File: Client/Frontend/Intro/OnboardingCardViewController.swift [hunk 2]
```diff
@@ -190,7 +200,7 @@ class OnboardingCardViewController: UIViewController, Themeable {
         // Adapt layout for smaller screens
         let scrollViewVerticalPadding = shouldUseSmallDeviceLayout ? UX.smallScrollViewVerticalPadding :  UX.scrollViewVerticalPadding
         let stackViewSpacingButtons = shouldUseSmallDeviceLayout ? UX.smallStackViewSpacingButtons :  UX.stackViewSpacingButtons
-        let imageViewHeight = shouldUseSmallDeviceLayout ?
+        let imageViewHeight = shouldUseTinyDeviceLayout ? UX.tinyImageViewSize.height : shouldUseSmallDeviceLayout ?
```
Comment (yoanarios) [inline Client/Frontend/Intro/OnboardingCardViewController.swift:203]: nit: The logic is simple but I think is a bit long and difficult to read I wonder if a separate func makes sense for readability

File: Client/Frontend/Intro/IntroViewController.swift [hunk 3]
```diff
@@ -202,7 +205,7 @@ extension IntroViewController: OnboardingCardDelegate {
         flowType: FxAPageType = .emailLoginFlow,
         referringPage: ReferringPage = .onboarding
     ) {
-        let singInSyncVC = FirefoxAccountSignInViewController.getSignInOrFxASettingsVC(fxaOptions,
+        let signInSyncVC = FirefoxAccountSignInViewController.getSignInOrFxASettingsVC(fxaOptions,
```
Comment (adudenamedruby) [inline Client/Frontend/Intro/IntroViewController.swift:208]: I cannot approve the removal of singing. :P

File: Client/Nimbus/NimbusFeatureFlagLayer.swift [hunk 4]
```diff
@@ -83,6 +84,17 @@ final class NimbusFeatureFlagLayer {
         }
     }
 
+    public func checkNimbusConfigForOnboardingNotificationCard(using nimbus: FxNimbus = FxNimbus.shared) -> OnboardingNotificationCardPosition {
```
Comment (adudenamedruby) [inline Client/Nimbus/NimbusFeatureFlagLayer.swift:87]: nit: I am bothered by how long this is. Is it inour 120 chars thing? I really need to get around to fixing that rule in Swiftlint. TOO MANY VIOLATIONS! lols

File: Client/OnboardingNotificationCardHelper.swift [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+class OnboardingNotificationCardHelper: FeatureFlaggable {
+    var cardPosition: OnboardingNotificationCardPosition {
+        return featureFlags.getCustomState(for: .onboardingNotificationCard) ?? .noCard
```
Comment (adudenamedruby) [inline Client/OnboardingNotificationCardHelper.swift:9]: How come this has a default, given that nimbus has a default? Theoretically, we shouldn't.

File: nimbus-features/onboardingFeature.yaml [hunk 6]
```diff
@@ -12,14 +12,32 @@ features:
...
-          "first-run-flow": true
+          "first-run-flow": true,
+          "notification-card-position": no-card
         }
       - channel: developer
         value: {
           "upgrade-flow": false,
-          "first-run-flow": true
+          "first-run-flow": true,
+          "notification-card-position": no-card
```
Comment (adudenamedruby) [inline nimbus-features/onboardingFeature.yaml:31]: should the dev default be `no-card`? Do we want to see something, at least? I have no deep thoughts re: this. Just asking.

File: Client/FeatureFlags/FeatureFlagsManager.swift [hunk 7]
```diff
@@ -107,16 +109,18 @@ class FeatureFlagsManager: HasNimbusFeatureFlags {
         let feature = NimbusFlaggableFeature(withID: convertCustomIDToStandard(featureID),
                                              and: profile)
         switch featureID {
-        case .startAtHome:
-            if let option = desiredState as? StartAtHomeSetting {
-                feature.setUserPreference(to: option.rawValue)
-            }
+        case .onboardingNotificationCard: return
```
[the remark this answers is not in the document]
Comment (thatswinnie) [reply Client/FeatureFlags/FeatureFlagsManager.swift:112]: Yes, it's not settable, however all feature flags with custom ids must be represented or otherwise we'll get a build error. That's why `wallpaperVersion` is in the switch as well. I am not sure throwing is a good option here - what benefit would it give us over ignoring the set call?

File: nimbus-features/onboardingFeature.yaml (hunk 6 above)
Comment (thatswinnie) [reply nimbus-features/onboardingFeature.yaml:31]: Yeah, we could definitely have something else as default for devs. Good idea.

File: Client/OnboardingNotificationCardHelper.swift (hunk 5 above)
Comment (adudenamedruby) [reply Client/OnboardingNotificationCardHelper.swift:9]: Yes, it does, doesn't it. Hmm. Good point. I totally forgot. :)

File: Client/Frontend/Intro/IntroViewModel.swift (hunk 1 above)
Comment (thatswinnie) [reply Client/Frontend/Intro/IntroViewModel.swift:52]: This is currently not testable. I created [a task](https://mozilla-hub.atlassian.net/browse/FXIOS-5893) to make Nimbus feature flags like this one testable.

File: Client/Experiments/Messaging/GleanPlumbMessageManager.swift [hunk 8]
```diff
@@ -115,7 +115,7 @@ class GleanPlumbMessageManager: GleanPlumbMessageManagerProtocol {
 
         TelemetryWrapper.recordEvent(category: .information,
                                      method: .view,
-                                     object: .homeTabBanner,
+                                     object: .messagingSurface,
```
Comment (adudenamedruby) [inline Client/Experiments/Messaging/GleanPlumbMessageManager.swift:118]: I'm curious as to why these changes are in here. These are already in `main` so... 🤷

**Review by adudenamedruby — APPROVED** [review]

File: Client/Experiments/Messaging/GleanPlumbMessageManager.swift (hunk 8 above)
Comment (thatswinnie) [reply Client/Experiments/Messaging/GleanPlumbMessageManager.swift:118]: That's an issue caused by the feature branch this merges in being behind this branch.

Comment (lmarceau) [reply Client/Experiments/Messaging/GleanPlumbMessageManager.swift:118]: epic branches "fun" 😆 1 more reason to avoid them I guess! Epic-branch should be kept to date with main as much as possible.

**Review by yoanarios — APPROVED** [review]
